### DISCUSSION TITLE
jbang and GitLab CI

### DISCUSSION THREAD

**Discussion by jmini** [jbangdev/jbang#1637] [Ideas]
I have started to use jbang scripts inside GitLab CI.

My current setup:

* jbang wrapper added to the repo
* Running inside an image providing java (I am using `eclipse-temurin:17`)
* I have set explicit values for `JBANG_CACHE_DIR` and `JBANG_REPO` (in order to share those folders between jobs by caching them)
* I have 2 jobs:
    * `build` is running on each commit (runs `jbang build`)
    * `execute` is a manually triggered job (runs `jbang execute`)

Complete example: https://github.com/jmini/jbang-ci-test

---

This is working great, but I was wondering why I can't used a docker image containing jbang (If I got it right, this would be `jbangdev/jbang-action`).
The Git Repo would contain only the script and the GitLab pipeline config…

But this attempt is failing with:

```
Using docker image sha256:efcae1d0877919e817568dfc5a3094df1b977a197ad53d9c036481b3171584d5 for jbangdev/jbang-action:0.106.1 with digest jbangdev/jbang-action@sha256:877ca9562cbff3a71fec38ea807088900a6da73f2291f6ad0414be66f422cf75 ...
jbang sh -c if [ -x /usr/local/bin/bash ]; then
	exec /usr/local/bin/bash 
elif [ -x /usr/bin/bash ]; then
	exec /usr/bin/bash 
elif [ -x /bin/bash ]; then
	exec /bin/bash 
elif [ -x /usr/local/bin/sh ]; then
	exec /usr/local/bin/sh 
elif [ -x /usr/bin/sh ]; then
	exec /usr/bin/sh 
elif [ -x /bin/sh ]; then
	exec /bin/sh 
elif [ -x /busybox/sh ]; then
	exec /busybox/sh 
else
	echo shell not found
	exit 1
fi
[jbang] [ERROR] Script or alias could not be found or read: 'sh'
[jbang] Run with --verbose for more details
Cleaning up project directory and file based variables
00:00
ERROR: Job failed: exit code 2
```

I am not sure to understand why, and I was also curious about differences between GitHub action and GitLab CI in this context…

**Comment by maxandersen** [comment]
How did you try run the jabang action container image ?

You'll need to do a volume mount into /ws to have it picked up. 

I believe github action does that automatically when used but not 100% sure.

**Reply by jmini** [reply]
I am not sure, but with the CI like GitLab-CI or CircleCI, the code of the CI pipeline (usually yaml steps) are directly running "inside" that container image.
So probably `jbangdev/jbang-action` can't be used for that…

**Reply by maxandersen** [reply]
sorry for delay in coming back to you on this one but I read https://docs.gitlab.com/ee/ci/docker/using_docker_images.html and it looks like it simply runs "inside" the container thus it should just work.

You might need to use absolute paths to `/builds/<project-path>/` or at least change the entrypoint of the image so it does not try to execute jbang directly.
